tests/measurements/test_mid_measure.py

```diff
@@ -32,57 +33,278 @@ def test_many_wires_error(self):
+        m2 = MeasurementValue(["m2"], lambda v: v)
+        compared = m1 >= m2
+        assert compared[0] is True
+        assert compared[1] is False
+        assert compared[2] is True
+        assert compared[3] is True
+
+    def test_equality_with_scalar(self):
+        """test the __eq__ dunder method between a MeasurementValue and an integer."""
+        m = MeasurementValue(["m"], lambda v: v)
+        m_eq = m == 0
+        assert m_eq[0] is True  # confirming value is actually eq to True, not just truthy
+        assert m_eq[1] is False
+
+    def test_equality_with_scalar_opposite(self):
+        """test the __eq__ dunder method between a MeasurementValue and an integer."""
+        m = MeasurementValue(["m"], lambda v: v)
+        m_eq = m == 1
+        assert m_eq[0] is False
+        assert m_eq[1] is True
+
+    def test_eq_with_other_measurement_value(self):
+        """test the __eq__ dunder method between two MeasurementValues."""
+        m1 = MeasurementValue(["m1"], lambda v: v)
+        m2 = MeasurementValue(["m2"], lambda v: v)
+        compared = m1 == m2
+        assert compared[0] is True
+        assert compared[1] is False
+        assert compared[2] is False
+        assert compared[3] is True
+
+    def test_merge_measurements_values_dependant_on_same_measurement(self):
+        """test that the _merge operation does not create more than 2 branches when combining two MeasurementValues
+        that are based on the same measurement."""
+        m0 = MeasurementValue(["m"], lambda v: v)
+        m1 = MeasurementValue(["m"], lambda v: v)
+        combined = m0 + m1
+        assert combined[0] == 0
+        assert combined[1] == 2
+
+    def test_combine_measurement_value_with_non_measurement(self):
+        """test that we can use dunder methods to combine a MeasurementValue with the underlying "primitive"
+        of that measurement value."""
+        m0 = MeasurementValue(["m"], lambda v: v)
+        out = m0 + 10
+        assert out[0] == 10
+        assert out[1] == 11
 
-    def test_measurement_value_assertion_error(self):
-        """Test that the return_type related info is updated for a
-        measurement."""
-        mv = MeasurementValue(measurement_id="1234")
+    def test_str(self):
+        """test that the output of the __str__ dunder method is as expected"""
+        m = MeasurementValue(["m"], lambda v: v)
+        assert str(m) == "if m=0 => 0\nif m=1 => 1"
 
-        with pytest.raises(MeasurementValueError, match="Unknown measurement value asserted"):
-            mv == -1
+    def test_complex_str(self):
+        """test that the output of the __str__ dunder method is as expected
+        w.r.t a more complicated MeasurementValue"""
+        a = MeasurementValue(["a"], lambda v: v)
+        b = MeasurementValue(["b"], lambda v: v)
+        assert (
+            str(a + b)
+            == """if a=0,b=0 => 0
+if a=0,b=1 => 1
+if a=1,b=0 => 1
+if a=1,b=1 => 2"""
+        )
```

Comment:
I agree with @antalszava some more complicated compositions tests would be great!